libusb/os/windows_winusb.c

```diff
@@ -1242,11 +1241,11 @@ static bool get_dev_port_number(HDEVINFO dev_info, SP_DEVINFO_DATA *dev_info_dat
 		// Check for the required format.
 		if (strncmp(buffer, "Port_#", 6) == 0) {
 			start = buffer + 6;
-			port = strtoll(start, &end, 10);
-			if (port < 0 || port > ULONG_MAX || end == start || *end != '\0') {
+			port = strtoul(start, &end, 10);
```

Comment:
The use of over-sized `strtoll` was deliberate, to catch and reject malformed things like negative numbers, or huge positive numbers. 

Could you also add a comment in the code itself about 0 being invalid please.

Reply:
Good point - my intuitive thought was to use the unsigned API where an unsigned number is needed, but `strtoll` will catch not just overflow but negative values as well here.

(Technically `strtoul` is supposed to be capable of this as well, since `errno` is supposed to be set to `ERANGE` in this case. But once again the MS CRT lets us down and doesn't actually do what the documentation says it does. Apart from that, the `strtoll` version is shorter and easier to read anyway.)

Added comment re: 0  being invalid.